### DISCUSSION TITLE
Style Selector to select `self`?

### DISCUSSION THREAD

**Discussion by WarWithinMe** [AvaloniaUI/Avalonia#11948] [Q&A] [answered]
I am trying to make a vertical TabStrip today, and this is my attempt:

```
<StackPanel Orientation="Horizontal">
    <StackPanel.Styles>
        <Style Selector="TabStrip /template/ ItemsPresenter#PART_ItemsPresenter > WrapPanel">
            <Setter Property="Orientation" Value="Vertical" />
         </Style>
    </StackPanel.Styles>
            
     <TabStrip ItemsSource="{Binding MainViewTabs}">
         <TabStrip.ItemTemplate>
              <DataTemplate x:DataType="vm:MainViewTabItem">
              <TextBlock Text="{Binding Text}"/>
              </DataTemplate>
          </TabStrip.ItemTemplate>
    </TabStrip>
    ...
</StackPanel>
```

I want to move the styling into TabStrip.Styles like so:
```
<TabStrip>
    <TabStrip.Styles>
         <Style Selector="/template/ ItemsPresenter#PART_ItemsPresenter > WrapPanel">
            <Setter Property="Orientation" Value="Vertical" />
         </Style>
    </TabStrip.Styles>
</TabStrip>
```
I remove `TabStrip` at the beginning from the Selector.
But it seems like Selector can not start with `/template/`.
It will work if I change it to `Selector="ItemsPresenter#PART_ItemsPresenter > WrapPanel"`,
but I think this is not very good as it might accidentally affect WrapPanel in the TabItem.

So what is the correct way to select only the WrapPanel of the TabStrip if the style is located inside the TabStrip control?

**Comment by timunie** [comment, marked as answer]
Which Avalonia verison? In general, please look up original styles. Most likely it's easiest to just set the ItemsPanelTemplate to whatever you need.

**Reply by WarWithinMe** [reply]
I'm using 11-rc1.1. Your suggestion is really a good way to go.
